radish34/docker-compose.yml

```diff
@@ -196,45 +241,249 @@ services:
+    depends_on:
+      - mongo-supplier1
+      - redis-supplier1
+      - geth-node
+    env_file:
+      - ./messenger/docker.env
+    build:
+      context: ./messenger
+      dockerfile: ./Dockerfile
+    environment:
+      VIRTUAL_HOST: messenger-supplier1.docker
+      MONGO_URL: mongodb://mongo-supplier1:27017/radish34
+      RADISH_API_URL: http://api-supplier1:8101
+      REDIS_URL: redis://redis-supplier1:6379
+    networks:
+      - network-supplier1
+    ports:
+     - 4002:4001
+    healthcheck:
+      test: ["CMD", "wget", "--spider", "-q", "http://localhost:4001/api/v1/health"]
+      interval: 30s
+      timeout: 10s
+      retries: 3
+      start_period: 30s
+
+  messenger-supplier2:
+    hostname: messenger-supplier2
+    container_name: messenger-supplier2
+    depends_on:
+      - mongo-supplier2
+      - redis-supplier2
+      - geth-node
+    env_file:
+      - ./messenger/docker.env
+    build:
+      context: ./messenger
+      dockerfile: ./Dockerfile
+    environment:
+      VIRTUAL_HOST: messenger-supplier2.docker
+      MONGO_URL: mongodb://mongo-supplier2:27017/radish34
+      RADISH_API_URL: http://api-supplier2:8101
+      REDIS_URL: redis://redis-supplier2:6379
+    networks:
+      - network-supplier2
+    ports:
+      - 4003:4001
+    healthcheck:
+      test: ["CMD", "wget", "--spider", "-q", "http://localhost:4001/api/v1/health"]
+      interval: 30s
+      timeout: 10s
+      retries: 3
+      start_period: 30s
 
   mongo-buyer:
     image: mongo:latest
     container_name: mongo-buyer
     volumes:
       - mongo-buyer:/data/db
+      - ./bin/mongo_health.sh:/health-check.sh:delegated
     logging:
       options:
         max-size: 10m
     networks:
       - network-buyer
     ports:
       - 27117:27017
+    healthcheck:
+      test: ["CMD", "sh", "health-check.sh"]
+      interval: 30s
+      timeout: 10s
+      retries: 3
+      start_period: 30s
 
   mongo-supplier1:
     image: mongo:latest
     container_name: mongo-supplier1
     volumes:
       - mongo-supplier1:/data/db
+      - ./bin/mongo_health.sh:/health-check.sh:delegated
     logging:
       options:
         max-size: 10m
     networks:
       - network-supplier1
     ports:
       - 27217:27017
+    healthcheck:
+      test: ["CMD", "sh", "health-check.sh"]
+      interval: 30s
+      timeout: 10s
+      retries: 3
+      start_period: 30s
 
   mongo-supplier2:
     image: mongo:latest
     container_name: mongo-supplier2
     volumes:
       - mongo-supplier2:/data/db
+      - ./bin/mongo_health.sh:/health-check.sh:delegated
     logging:
       options:
         max-size: 10m
     networks:
       - network-supplier2
     ports:
       - 27317:27017
+    healthcheck:
+      test: ["CMD", "sh", "health-check.sh"]
+      interval: 30s
+      timeout: 10s
+      retries: 3
+      start_period: 30s
+
+  redis-buyer:
+    image: redis
+    container_name: redis-buyer
+    networks:
+      - network-buyer
+    ports:
+      - 6379
```

Comment:
Can we expose this on localhost so that the messenger service can still connect to it if messenger is running as a local nodejs process instead of in the docker environment?
`6379` --> `6379:6379`

Reply:
okay, can do.

Reply:
So, I can make that change but this is how it was before. I've just moved some of the services around in the file so it looks like I've done this. If you would like this change, what do you propose I do for the other redis containers that are on the same port but each on their own networks?

Reply:
Hmmm this is what I see on current master branch: https://github.com/ethereum-oasis/baseline/blob/master/radish34/docker-compose.yml#L323. The `messenger` service-level tests only need one user and therefore only one redis container, so we only need to expose the port on that one container to make that test suite happy

Reply:
The `messenger` test suite still passes, it just produces this new error:
```
2020-04-17 12:37:19 [debug]: 	SUCCESS: connected to bull queue "baseline:messenger:receiveMessage:req" at redis://localhost:6379 
  console.error node_modules/jest-jasmine2/build/jasmine/Env.js:248
    Unhandled error

  console.error node_modules/jest-jasmine2/build/jasmine/Env.js:249
    Error: connect ECONNREFUSED 127.0.0.1:6379
        at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1137:16)
```

Reply:
Okay, Looks like I edited the line then. 🤷 I'll return it to what it was